test-suite/js/src/test/require-modules/org/scalajs/testsuite/jsinterop/ModulesTest.scala

```diff
@@ -118,4 +122,13 @@ object ModulesTest {
 
     def isBuffer(x: Any): Boolean = js.native
   }
+
+  // #4001 - Test that unused super-classes are not imported.
+  @js.native
+  @JSImport("non-existent", "Foo")
+  class NonExistentSuperClass extends js.Object
+
+  @js.native
+  @JSImport("string_decoder", "StringDecoder")
+  class ExistentSubClass extends NonExistentSuperClass
```

Comment:
Should we also add a test case with an `object ExistentObject extends NonExistentSuperClass`?